cli/containerizedengine/progress.go

```diff
@@ -0,0 +1,208 @@
+package containerizedengine
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/containerd/containerd/content"
+	"github.com/containerd/containerd/errdefs"
+	"github.com/containerd/containerd/remotes"
+	"github.com/docker/docker/pkg/jsonmessage"
+	digest "github.com/opencontainers/go-digest"
+	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
+	"github.com/sirupsen/logrus"
+)
+
+// nolint: gocyclo
```

Comment:
this code is migrated from another repo with slight tweaks to use docker CLI progress reporting instead of ctr progress reporting - I was trying not to significantly restructure it, but I'll see if I can do some limited refactoring without messing things up.